qgeologis/data_interface.py

```diff
@@ -127,16 +146,33 @@ def __build_data(self):
         if self.__uom is not None and self.__uom.startswith("@"):
             req2 = QgsFeatureRequest(req)
             req2.setLimit(1)
-            for f in self.__layer.getFeatures(req):
+            for f in self.__layer.getFeatures(req2):
                 self.__uom = f[self.__uom[1:]]
                 break
 
         req.setSubsetOfAttributes([self.__x_fieldname, self.__y_fieldname], self.__layer.fields())
         # Do not forget to add an index on this field to speed up ordering
         req.addOrderBy(self.__x_fieldname)
 
-        xy_values = [(f[self.__x_fieldname], f[self.__y_fieldname] if f[self.__y_fieldname] is not None else self.__nodata_value)
-                     for f in self.__layer.getFeatures(req)]
+        def is_null(v):
+            return isinstance(v, QVariant) and v.isNull()
+
+        if self.__nodata_value is not None:
+            # replace null values by a 'Nodata' value
+            xy_values = [(
+                f[self.__x_fieldname], f[self.__y_fieldname]
+                if not is_null(f[self.__y_fieldname])
+                else self.__nodata_value
+            )
+                         for f in self.__layer.getFeatures(req)
```

Comment:
Not sure understanding why you use `req` there, whilst this is `req2` in line 149. Is it ok for this variable?

Reply:
req2 has a limit(1) set on it, just to query for the first feature in case we do not know the unit of measure.

But I admit the name is misleading, I will rename it, thanks :)